File: backend/app/schemas/section.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime

class SectionBase(BaseModel):
    """Базовая схема отдела"""
    name: str
    code: str
    division_id: int
    description: Optional[str] = None
    is_active: bool = True
    
class SectionCreate(SectionBase):
    """Схема для создания отдела"""
    pass
    
class SectionUpdate(BaseModel):
    """Схема для обновления отдела"""
    name: Optional[str] = None
    code: Optional[str] = None
    division_id: Optional[int] = None
    description: Optional[str] = None
    is_active: Optional[bool] = None
    
class SectionInDBBase(SectionBase):
    """Базовая схема для отдела в БД"""
    id: int
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True
        
class Section(SectionInDBBase):
    """Схема для возврата отдела через API"""
    pass 

File: frontend/src/types/section.ts
import { Division } from './division'; // Импортируем тип Division, если нужно

export interface Section {
  id: number;
  name: string;
  code: string;
  division_id: number;
  description: string | null;
  is_active: boolean;
  created_at: string; // Или Date
  updated_at: string; // Или Date
  division?: Division; // Опционально, если нужна информация о подразделении
}

export interface SectionCreate {
  name: string;
  code: string;
  division_id: number;
  description?: string | null;
  is_active?: boolean;
}

export type SectionUpdate = Partial<SectionCreate>; 